### PR TITLE
Add `Buffer`, a common dependency, to global scope

### LINKED ISSUE #34: Include Buffer polyfill by default
A common reported issue is that third party libraries do not work for multiple reasons.

A very common solution is to include Node's `Buffer` polyfill.

Because people think it's an error on our side, instead of their libraries, we should handle the common case.

While we shouldn't include Buffer in the core sandbox, we should include it in the templates.

## Outcome

Add Buffer polyfill to template monorepo build system

### PR DESCRIPTION
**Pull request by ziad-saab**
This PR adds `Buffer` to the global scope by importing it from a polyfill. Fixes #34

### CODE REVIEW

**Comment by ziad-saab** [comment]
Why we can't just add the line `globalThis.Buffer = xxx` to the `index.ts` of the snap:

`import`s are evaluated "statically" in theory. Even in transpiled code they're executed first, even before source code that is "above" them in the module. By putting the `globalThis` code in its own module, and loading that module first, we ensure that `Buffer` is available everywhere, including in other `import`s from `index.ts`.

File: packages/snap/src/buffer-fix.ts
Comment (Mrtenz) [inline packages/snap/src/buffer-fix.ts:1]: Have you verified that this works with dependencies using `Buffer`?

**Comment by ritave** [comment]
> Why we can't just add the line `globalThis.Buffer = xxx` to the `index.ts` of the snap:
> 
> `import`s are evaluated "statically" in theory. Even in transpiled code they're executed first, even before source code that is "above" them in the module. By putting the `globalThis` code in its own module, and loading that module first, we ensure that `Buffer` is available everywhere, including in other `import`s from `index.ts`.

Adding this as a comment to the fix file will be cool

File: packages/snap/src/index.ts [hunk 1]
```diff
@@ -1,3 +1,9 @@
+/**
+ * This side-effect import puts Buffer in the global scope before
+ * any other module or snap code is loaded.
+ */
+// eslint-disable-next-line import/no-unassigned-import
+import './buffer-fix';
```
Comment (FrederikBolding) [inline packages/snap/src/index.ts:6]: Is it cleaner to do this in the `snap.config.js` so we don't have to have this in our source code?

e.g.

```
module.exports = {
  cliOptions: {
    src: './src/index.ts',
    port: 8080,
  },
  bundlerCustomizer: (bundler) => {
    const through = require('through2');
    bundler.transform(function () {
      let data = '';
      return through(
        function (buf, _enc, cb) {
          data += buf;
          cb();
        },
        // eslint-disable-next-line consistent-return
        function (cb) {
          this.push("globalThis.Buffer = require('buffer/').Buffer;");
          this.push(data);
          cb();
        },
      );
    });
  },
};
```

File: packages/snap/snap.config.js [hunk 2]
```diff
@@ -3,4 +3,23 @@ module.exports = {
     src: './src/index.ts',
     port: 8080,
   },
+  bundlerCustomizer: (bundler) => {
+    // eslint-disable-next-line node/global-require
+    const through = require('through2');
```
Comment (Mrtenz) [inline packages/snap/snap.config.js:8]: Couldn't this just be imported at the top of this file?

File: packages/snap/snap.config.js [hunk 3]
```diff
@@ -3,4 +3,23 @@ module.exports = {
...
+    bundler.transform(function () {
+      let data = '';
+      return through(
+        function (buf, _enc, cb) {
```
Comment (Mrtenz) [inline packages/snap/snap.config.js:12]:
```suggestion
        function (buffer, _encoding, callback) {
```

File: packages/snap/snap.config.js [hunk 4]
```diff
@@ -3,4 +3,23 @@ module.exports = {
...
+          data += buf;
+          cb();
+        },
+        // eslint-disable-next-line consistent-return
```
Comment (Mrtenz) [inline packages/snap/snap.config.js:16]: Is this needed?

**Review by Mrtenz — CHANGES_REQUESTED** [review]

File: packages/snap/snap.config.js [hunk 5]
```diff
@@ -1,6 +1,24 @@
...
+      let data = '';
+      return through(
+        function (buffer, _encoding, callback) {
+          data += buffer;
+          callback();
+        },
+        function (cb) {
+          this.push("globalThis.Buffer = require('buffer/').Buffer;");
+          this.push(data);
+          cb();
```
Comment (Mrtenz) [inline packages/snap/snap.config.js:19]:
```suggestion
        function (callback) {
          this.push("globalThis.Buffer = require('buffer/').Buffer;");
          this.push(data);
          callback();
```

**Review by Mrtenz — CHANGES_REQUESTED** [review]

**Review by Mrtenz — APPROVED** [review]

**Pull request merged**